pkg/reconciler/shared/tektonconfig/tektonconfig.go

```diff
@@ -160,6 +161,18 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonConfi
 		}
 		return nil
 	}
+	if !tc.Spec.TektonPruner.Disabled {
+		tektonPruner := pruner.GetTektonPrunerCR(tc, r.operatorVersion)
+		if _, err = pruner.EnsureTektonPrunerExists(ctx, r.operatorClientSet.OperatorV1alpha1().TektonPruners(), tektonPruner); err != nil {
+			tc.Status.MarkComponentNotReady(fmt.Sprintf("TektonPruner %s", err.Error()))
+			return v1alpha1.REQUEUE_EVENT_AFTER
+		}
+	} else {
+		if err := pruner.EnsureTektonPrunerCRNotExists(ctx, r.operatorClientSet.OperatorV1alpha1().TektonPruners()); err != nil {
+			tc.Status.MarkComponentNotReady(fmt.Sprintf("TektonPruner: %s", err.Error()))
+			return v1alpha1.REQUEUE_EVENT_AFTER
+		}
+	}
```

Comment:
I think this is fine, but we should discuss this once if we should make it a part of TektonConfig